File: app/src/main/java/com/example/league/mobilecodingchallenge/main/MainActivity.java
package com.example.league.mobilecodingchallenge.main;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import com.example.league.mobilecodingchallenge.R;
import com.example.league.mobilecodingchallenge.api.OnRequestCallback;
import com.example.league.mobilecodingchallenge.api.Service;
import com.example.league.mobilecodingchallenge.model.Account;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onResume() {
        super.onResume();

        Service.login("Hello", "World", new OnRequestCallback<Account>() {
            @Override
            public void onSuccess(Account response) {
                Log.v(TAG, response.getApiKey());
            }

            @Override
            public void onFailure(String errorMessage) {
                Log.e(TAG, errorMessage);
            }
        });
    }

}

File: kotlin_app/src/main/java/life/league/challenge/kotlin/main/MainActivity.kt
package life.league.challenge.kotlin.main

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import life.league.challenge.kotlin.R
import life.league.challenge.kotlin.api.Either
import life.league.challenge.kotlin.api.Service


class MainActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onResume() {
        super.onResume()

//        Service.login("Hello", "World") { result ->
//            when (result) {
//                is Either.Success -> Log.v(TAG, result.response?.apiKey ?: "")
//                is Either.Failure -> Log.e(TAG, result.errorResponse)
//            }
//        }
    }

}
